cmd/ctr/commands/shim/pprof.go

```diff
@@ -0,0 +1,165 @@
+//go:build !windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package shim
+
+import (
+	gocontext "context"
+	"errors"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/containerd/containerd/v2/cmd/ctr/commands/pprof"
+	"github.com/containerd/containerd/v2/pkg/namespaces"
+	"github.com/containerd/containerd/v2/pkg/shim"
+	"github.com/urfave/cli/v2"
+)
+
+var pprofCommand = &cli.Command{
+	Name:  "pprof",
+	Usage: "Provide golang pprof outputs for containerd-shim",
+	Subcommands: []*cli.Command{
+		pprofBlockCommand,
+		pprofGoroutinesCommand,
+		pprofHeapCommand,
+		pprofProfileCommand,
+		pprofThreadcreateCommand,
+		pprofTraceCommand,
+	},
+}
+
+var pprofGoroutinesCommand = &cli.Command{
+	Name:  "goroutines",
+	Usage: "Dump goroutine stack dump",
+	Flags: []cli.Flag{
+		&cli.UintFlag{
+			Name:  "debug",
+			Usage: "Debug pprof args",
+			Value: 2,
+		},
+	},
+	Action: func(context *cli.Context) error {
+		return pprof.GoroutineProfile(context, getPProfClient)
+	},
+}
+
+var pprofHeapCommand = &cli.Command{
+	Name:  "heap",
+	Usage: "Dump heap profile",
+	Flags: []cli.Flag{
+		&cli.UintFlag{
+			Name:  "debug",
+			Usage: "Debug pprof args",
+			Value: 0,
+		},
+	},
+	Action: func(context *cli.Context) error {
+		return pprof.HeapProfile(context, getPProfClient)
+	},
+}
+
+var pprofProfileCommand = &cli.Command{
+	Name:  "profile",
+	Usage: "CPU profile",
+	Flags: []cli.Flag{
+		&cli.DurationFlag{
+			Name:    "seconds",
+			Aliases: []string{"s"},
+			Usage:   "Duration for collection (seconds)",
+			Value:   30 * time.Second,
```

Comment:
should it be 30 here or `30 * time.Second`?

Reply:
it should be `30 * time.Second`. The flag here has the type `cli.DurationFlag`. See code sample here: https://cli.urfave.org/v2/examples/full-api-example/